minecode_pipelines/pipes/composer.py

```diff
@@ -0,0 +1,95 @@
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# Data Generated with ScanCode.io is provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+# ScanCode.io should be considered or used as legal advice. Consult an Attorney
+# for any legal advice.
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/aboutcode-org/scancode.io for support and download.
+
+from pathlib import Path
+from aboutcode import hashid
+from aboutcode.hashid import get_package_base_dir
+from minecode_pipelines.miners.composer import get_composer_packages
+from minecode_pipelines.miners.composer import load_composer_packages
+from minecode_pipelines.miners.composer import get_composer_purl
+from minecode_pipelines.pipes import write_data_to_yaml_file
+from scanpipe.pipes.federatedcode import commit_changes
+from scanpipe.pipes.federatedcode import push_changes
+from minecode_pipelines import VERSION
+
+PACKAGE_BATCH_SIZE = 1000
+
+
+def mine_composer_packages():
+    """Mine Composer package names from Packagist and return List of (vendor, package) tuples."""
+    packages_file = get_composer_packages()
+    return load_composer_packages(packages_file)
+
+
+def mine_and_publish_composer_purls(packages, cloned_data_repo, logger):
+    """Mine Composer packages and publish their PURLs to a FederatedCode repository."""
+    counter = 0
+    purl_files = []
+    purls = []
+    for vendor, package in packages:
+        logger(f"getting packageURLs for package: {vendor}/{package}")
+
+        updated_purls = get_composer_purl(vendor, package)
+        if not updated_purls:
+            continue
+
+        base_purl = updated_purls[0]
+        package_base_dir = get_package_base_dir(purl=base_purl)
+
+        logger(f"writing packageURLs for package: {base_purl} at: {package_base_dir}")
+        purls_string = " ".join(updated_purls)
+        logger(f"packageURLs: {purls_string}")
+
+        purl_file_full_path = Path(
+            cloned_data_repo.working_dir
+        ) / hashid.get_package_purls_yml_file_path(base_purl)
+
+        write_data_to_yaml_file(path=purl_file_full_path, data=updated_purls)
+
+        purl_files.append(purl_file_full_path)
+        purls.append(str(package_base_dir))
+        counter += 1
+
+        if counter >= PACKAGE_BATCH_SIZE:
+            commit_changes(
+                repo=cloned_data_repo,
+                files_to_commit=purl_files,
+                purls=purls,
+                mine_type="packageURL",
+                tool_name="pkg:pypi/minecode-pipelines",
+                tool_version=VERSION,
+            )
+            push_changes(repo=cloned_data_repo)
+            counter = 0
+            purl_files = []
```

Comment:
Why not initialize these inside the loop at start instead of so you don't have this code in 2 places?